src/test/java/com/example/moviemanager/service/MovieServiceTest.java

```diff
@@ -0,0 +1,155 @@
+package com.example.moviemanager.service;
+
+import com.example.moviemanager.exception.MovieNotFoundException;
+import com.example.moviemanager.repository.model.Movie;
+import com.example.moviemanager.repository.MovieRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class MovieServiceTest {
+
+    @Mock
+    private MovieRepository movieRepository;
+
+    @InjectMocks
+    private MovieService movieService;
+
+    @Test
+    void returns_list_of_all_movies() {
+        // given
+        List<Movie> expectedMovies = List.of(
+                new Movie(1, "Home Alone", "Christmas movie", 1990, 8.5),
+                new Movie(2, "Home Alone 2", "Christmas movie", 1992, 8.9));
+        when(movieRepository.findAll()).thenReturn(expectedMovies);
+
+        // when
+        List<Movie> actualMovies = movieService.findAll();
+
+        // then
+        assertEquals(expectedMovies, actualMovies);
+    }
+
+    @Test
+    void throws_exception_when_trying_to_find_non_existing_movie() {
+        // given
+        int movieId = 2;
+        String expectedMessage = "Could not find Movie by id - " + movieId;
+        when(movieRepository.findById(2)).thenThrow(new MovieNotFoundException(("Could not find Movie by id - " + movieId)));
+
+        // when
+        MovieNotFoundException thrown =
+                assertThrows(MovieNotFoundException.class, () -> movieService.findById(2));
+        String actualMessage = thrown.getMessage();
+
+        // then
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    void returns_exactly_one_movie() {
```

Comment:
I think you should not emphasize that it is "exactly one movie" because it is clear even from method signature, but rather what is bussines case. "finds movie by id"